webpack/assets/javascripts/react_app/redux/API/APIReducer.js

```diff
@@ -18,11 +18,13 @@ const apiReducer = (state = initialState, { type, key, payload, response }) => {
           ...state[key],
           payload,
           status: PENDING,
+          isFirstRequest: !state[key],
```

Comment:
I'm not sure what is the purpose for this, why do we need to know if it is the first request? the key exists in all of the following requests, not just the first one.

Reply:
for example if you want to do something like showing loading skeleton only on the first API request ( for example with polling )
and avoid showing the skeleton on the next polling requests.

you will be able to do this easily by using the selector `selectIsFirstRequest(state, key)`

Reply:
you can invoke that API request with react `componentDidMount` lifecycle or via `useEffect` hook  (with empty `[]`) instead so it will occur just once at the first rendering.

Reply:
IMO as a developer consumer I would be more happy to receive a "shortcut" instead of creating all of the lifecycle logic

Reply:
with too many shortcuts it can be a mess :) 
I do think in this case it makes sense to stick with react best practices